### PR TITLE
feat: detect flag extinctions

### PR DESCRIPTION
**Pull request by jazanne**
For each removed flag, we will check if there are any other references remaining i the code! This will make cleanup much easier

Learn more about [extinct flags](https://docs.launchdarkly.com/home/code/code-references/?q=extinct#understanding-extinction-events)

<details>
<summary>
Open to see design options
</summary>
### :x: 1 flag removed

Table options:

Option 1:

| Name | Key | Aliases found |
| --- | --- | --- |
| [Show widgets](https://app.launchdarkly.com/demo-dan-042021-2/development/features/show-widgets) | `show-widgets` | `showWidgets` |
| :skull: [Beta UI](https://app.launchdarkly.com/demo-dan-042021-2/development/features/beta-ui) | `beta-ui` | `betaUi`, `beta_ui` |
 
 

Option 2:

| Name | Key | Aliases found |
| --- | --- | --- |
| [Show widgets](https://app.launchdarkly.com/demo-dan-042021-2/development/features/show-widgets) | `show-widgets` | `showWidgets` |
| ✅ [Beta UI](https://app.launchdarkly.com/demo-dan-042021-2/development/features/beta-ui) (no remaining references) | `beta-ui` | `betaUi`, `beta_ui` |

Option 3:

| Name | Key | Aliases found | All references removed |
| --- | --- | --- | --- |
| [Show widgets](https://app.launchdarkly.com/demo-dan-042021-2/development/features/show-widgets) | `show-widgets` | `showWidgets` | false |
| [Beta UI](https://app.launchdarkly.com/demo-dan-042021-2/development/features/beta-ui) | `beta-ui` | `betaUi`, `beta_ui` | true |

Option 4:

| Name | Key | Aliases found | All references removed |
| --- | --- | --- | --- |
| [Show widgets](https://app.launchdarkly.com/demo-dan-042021-2/development/features/show-widgets) | `show-widgets` | `showWidgets` | |
| [Beta UI](https://app.launchdarkly.com/demo-dan-042021-2/development/features/beta-ui) | `beta-ui` | `betaUi`, `beta_ui` | ✅ |
</details>

### CODE REVIEW

**Comment by ld-kyee** [comment]
I think this is great! Although, I'm not sure the skull is very clear.  Maybe we need another column that says "all references removed" or something like this.

**Comment by UPatel12** [comment]
confirming that if there are no remaining references then the `alias found` colmn should be empty, right?

**Comment by jazanne** [comment]
> confirming that if there are no remaining references then the `alias found` colmn should be empty, right?

@UPatel12  No, the alias found column is showing that those aliases were found and removed - will tag you in a comment to make it more clear

File: testdata/test [hunk 1]
```diff
@@ -1,11 +1,7 @@
 show-widgets
 showWidgets
-betaUi
```
Comment (jazanne) [inline testdata/test:3]: this alias of `beta-ui` is what was removed, so we try to let people know

Comment (UPatel12) [reply testdata/test:3]: I'm leaning toward option 2 (without the check mark) out of the two options. The skull & check mark are a little confusing. 

I like Kerrie's idea of adding a new column for all references that have been removed.

2 options:
- Column can be True or False. Are all references removed or not. 
- Column can be a count of the references found/remaining. For extinct flags, the colmn entry would be zero.

**Comment by UPatel12** [comment]
I'm leaning toward option 2 (without the check mark) out of the two options. The skull & check mark are a little confusing. 

I like Kerrie's idea of adding a new column for all references that have been removed.

2 options:
- Column can be True or False. Are all references removed or not. 
- Column can be a count of the references found/remaining. For extinct flags, the colmn entry would be zero.

**Comment by jazanne** [comment]
> Column can be a count of the references found/remaining. For extinct flags, the colmn entry would be zero.

we don't scan for all flags in repo so we can't do count

I really think a boolean column w/true or false will not be visually compelling enough

Maybe the column is something like "all references removed" and then has a checkmark

| Name | Key | Aliases found | All references remove |
| --- | --- | --- | --- |
| [Show widgets](https://app.launchdarkly.com/demo-dan-042021-2/development/features/show-widgets) | `show-widgets` | `showWidgets` | |
| [Beta UI](https://app.launchdarkly.com/demo-dan-042021-2/development/features/beta-ui) | `beta-ui` | `betaUi`, `beta_ui` | ✅ |

**Comment by jazanne** [comment]
Proposal for change to 4 columns...


## LaunchDarkly flag references

### 🔍 2 flags added or modified

| Flag name | Key | Aliases found | Info |
| --- | --- | --- | --- |
| [Show: Live Tiles](https://app.launchdarkly.com/demo-dan-042021-2/development/features/show-widgets) | `show-widgets` | `showWidgets` |
| [old-pricing-banner](https://app.launchdarkly.com/demo-dan-042021-2/development/features/old-pricing-banner) | `old-pricing-banner` | `old-pricing-banner` |  ⚠️ archived on 2023-08-03 |


### ❌ 1 flag removed

| Flag name | Key | Aliases found | Info |
| --- | --- | --- | -- |
| [Show: App Promo](https://app.launchdarkly.com/demo-dan-042021-2/development/features/mobile-app-promo-ios) | `mobile-app-promo-ios` | `mobileAppPromoIos` | ℹ️ archived on 2023-08-03 |
| [Beta UI](https://app.launchdarkly.com/demo-dan-042021-2/development/features/beta-ui) | `beta-ui` | `betaUi`, `beta_ui` | ✅ all references removed |

File: main.go [hunk 2]
```diff
@@ -161,22 +174,21 @@ func getOptions(config *lcr.Config) (options.Options, error) {
...
-	flagsRemoved := make([]string, 0, len(flagsRef.FlagsRemoved))
-	for k := range flagsRef.FlagsRemoved {
-		flagsRemoved = append(flagsRemoved, k)
+	flagsRemoved := flagsRef.RemovedKeys()
+	setOutputsForChangedFlags("removed", flagsRemoved)
+
+	if config.CheckExtinctions {
+		setOutputsForChangedFlags("extinct", flagsRef.ExtinctKeys())
 	}
-	setOutputsForChangedFlags("removed", flagsModified)
```
Comment (jazanne) [inline main.go:175]: sad, this was wrong before

**Review by karayusuf — APPROVED** [review]

**Pull request merged**